_posts/2019-03-04-jumping-on-the-dash-less-bandwagon.markdown

```diff
@@ -0,0 +1,64 @@
+---
+layout: post
+title:  "Jumping on the '-less' Bandwagon"
+date:   2019-03-04
+tags: serverless
+author: Pete Yandell
+image: img/bandwagonless/bandwagonless.png
+---
+
+<img style="display: block; margin-left: auto; margin-right: auto; width: 40%;" alt="wagon wheel" src="{{ site.url }}/img/bandwagonless/bandwagonless.png"/>
+
+## Introduction
+
+It has been awhile since I made my [last post](https://mechanicalrock.github.io/2018/08/27/inception-pipelines-pt6), the one where at the end I promised something about the [AWS CDK](https://awslabs.github.io/aws-cdk/) and how awesome it was going to be. Since then a few of my colleagues have mastered it, and I got super excited by something else.
+
+That something else is SER-VER-LESS!! And so now, I to, am _Jumping on the '-less' Bandwagon!_
+
+## What is serverless and why all the hype?
+
+If you scour the Internet, you will find as many definitions as there are search results. Most of them dive into esoteric technical definitions of what is and isn’t serverless.
+
+My technical perspective is that I don’t consider building, running and managing containers as serverless. I also don't count no-infrastructure pieces of your architecture like DynamoDB, Identity management, payment gateways, etc as Serverless either. They are however critical ‘no-ops’ pieces to operating Serverlessly. What’s left is really the functions-as-a-service services like AWS Lambda, Google Functions and Azure Functions. These functions strip away everything that are not critical core features that are unique to your business (versus structural piece like databases)
```

Comment:
Thoughts -

I'd challenge "building containers" - this is a package management concern that none of the "serverless" offerings do out of the box. It's delegated to the framework (serverless, sam etc) or the language (static, dynamic) to handle. If your adding dependencies in a zip file, that is still package management - you are just vendor-ing your dependencies. If the container building process breaks your definition of serverless - technically lambda isn't serverless.

This is part of the reason why I prefer sticking to the execution/operational model/environment as the marker of serverless - and not the development cycle. And at that point - supporting infrastructure like IAM becomes critical.

Functions by themselves aren't useful - you need an event driven invocation model as well, which requires an event grid of some description and event sources. 

I know it's server-(less), but if you want to move this argument along one has to define something but what it is, rather than what it isn't.

Reply:
I like the challenging of the phrase "building containers" and how it is package management. I agree with you there. I would say the distinct though is size, which is the "execution/operational model/environment" marker. I've re-written this paragraph to read:

> My technical perspective is that I don’t consider writing full-sized applications and deploying them in containers as serverless. I also don't count no-infrastructure pieces of your architecture like DynamoDB, Identity management, payment gateways, etc as Serverless either. They are however critical ‘no-ops’ pieces to operating Serverlessly. What’s left are the functions-as-a-service services like AWS Lambda, Google Functions and Azure Functions. These [event-driven](https://en.wikipedia.org/wiki/Event-driven_architecture) functions strip away everything that are not critical core features which are unique to your business (versus structural piece like databases).